src/main/java/org/jenkins_ci/plugins/flexible_publish/ConditionalPublisher.java

```diff
@@ -177,14 +178,32 @@ public boolean prebuild(final AbstractBuild<?, ?> build, final BuildListener lis
         return true;
     }
 
+    private static String getBuildStepName(BuildStep s) {
+        if (s instanceof Describable) {
+            return String.format("%s (%s)", ((Describable<?>)s).getDescriptor().getDisplayName(), s.toString());
+        } else {
+            return s.toString();
+        }
+    }
+
     public boolean perform(final AbstractBuild<?, ?> build, final Launcher launcher, final BuildListener listener)
                                                                                                 throws InterruptedException, IOException {
+        // do as AbstractBuild.AbstractRunner#performAllBuildSteps
+        boolean wholeResult = true;
         for (BuildStep publisher: getPublisherList()) {
-            if (!runner.perform(condition, publisher, build, launcher, listener)) {
-                return false;
```

Comment:
> After your change it will execute all Actions that is wrong.

Let me see.
"all Actions" mean actions in the same `ConditionalPublisher`.
If you set "Don't run and fail", the condition will be evaluated for each actions and none won't run.